src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -1460,30 +1570,78 @@ void doDocument(DocumentManagement documentManagement,
             new Object[] { opentextDocId, fileDataSize });
         return;
     }
-    long versionNumber = version.getNumber();
-    String contextId = documentManagement.getVersionContentsContext(
-        documentNode.getID(), versionNumber);
-    ContentService contentService =
-        this.soapFactory.newContentService(documentManagement);
-    DataHandler dataHandler = contentService.downloadContent(contextId);
-
     String contentType = version.getMimeType();
     if (contentType != null) {
       response.setContentType(contentType);
     }
-    InputStream inputStream = dataHandler.getInputStream();
-    try {
-      if (inputStream != null) {
-        IOHelper.copyStream(inputStream, response.getOutputStream());
+
+    switch (this.downloadMethod) {
+      case "contentserver": {
```

Comment:
I know this is the default, but if you put this last, then it's easier to see the similarities of both the cws/cwscontenthandler and cwscontenthandler/contentserver pairs.